addons/account/models/account_payment.py

```diff
@@ -84,7 +94,16 @@ class account_payment(models.Model):
         string='Journal Item Label',
         help='Change label of the counterpart that will hold the payment difference',
         default='Write-Off')
-    partner_bank_account_id = fields.Many2one('res.partner.bank', string="Recipient Bank Account", readonly=True, states={'draft': [('readonly', False)]}, domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]")
+    possible_bank_partner_ids = fields.Many2many('res.partner', compute='_compute_possible_bank_partners')
+    partner_bank_account_id = fields.Many2one(
+        'res.partner.bank', string="Recipient Bank Account", readonly=True,
+        states={'draft': [('readonly', False)]},
+        # FIXME: should it be "if partner set then use that else use company"?
+        domain="""[
+            '|', ('company_id', '=', False), ('company_id', '=', company_id),
+            ('partner_id', 'in', possible_bank_partner_ids)
```

Comment:
Possibly less trashy / weird if we just add a commercial_partner_id field instead of a possible_bank_partner_ids: original is simple
```python
        return {'domain': {'partner_bank_account_id': [('partner_id', 'in', [self.partner_id.id, self.partner_id.commercial_partner_id.id])]}}
```
so seems unnecessary to add a full computed field rather than just a `related="partner_id.commercial_partner_id"`.

Reply:
Although do we want to allow anything or nothing if `partner_id` is not set? Original allows all until partner_id is modified (possibly odd behaviour if partner_id is removed given it's not required) also partner_id is ~ always set as it's returned by default_get.

Reply:
makes sense to always filter on partner for view concerns, model does not require because can have internal transfers